kfs-core/src/main/java/edu/arizona/kfs/sys/batch/service/impl/BuildingImportServiceImpl.java

```diff
@@ -220,75 +218,58 @@ private boolean isValid(ArchibusBuildings archBuilding, PrintWriter outReportErr
+			!isAttributeValid(isBuildingAddressStateCodeRequired, archBuilding.getBuildingAddressStateCode(), archBuilding.getBuildingAddressStateCode().length(), maxBuildingAddressStateCodeLen, reportMsg, outReportErrorWriter, "BuildingAddressStateCode(") || 
+			!isAttributeValid(isBuildingAddressZipCodeRequired, archBuilding.getBuildingAddressZipCode(), archBuilding.getBuildingAddressZipCode().length(), maxBuildingAddressZipCodeLen, reportMsg, outReportErrorWriter, "BuildingAddressZipCode(") || 
+			!isAttributeValid(isBuildingAddressCountryCodeRequired, archBuilding.getBuildingAddressCountryCode(), archBuilding.getBuildingAddressCountryCode().length(), maxBuildingAddressCountryCodeLen, reportMsg, outReportErrorWriter, "BuildingAddressCountryCode(")) {
+
+			return false;
 		}
-		if (isAttributeLenToLong(maxBuildingAddressZipCodeLen, archBuilding.getBuildingAddressZipCode().length())) {
+		return true;
+	}
+
+	private Boolean isAttributeValid(Boolean isAttributeRequired, String attributeValue, Integer attributeLen, Integer maxAttributeLen, String reportMsg, PrintWriter outReportErrorWriter, String validationMessage) {
+		boolean valid = true;
+		if ((attributeValue.equalsIgnoreCase(KFSParameterKeyConstants.HYPHEN) || StringUtils.isBlank(attributeValue)) && isAttributeRequired) {
+			reportMsg += validationMessage.concat(attributeValue) + ") is required, ";
 			valid = false;
-			reportMsg += "BuildingAddressZipCode(" + archBuilding.getBuildingAddressZipCode() + ") is longer than " + maxBuildingAddressZipCodeLen + ", ";
 		}
-		if (isAttributeNotValid(isBuildingAddressCountryCodeRequired, archBuilding.getBuildingAddressCountryCode())) {
-			valid = false; 
-			reportMsg += "BuildingAddressCountyCode(" + archBuilding.getBuildingAddressCountryCode() + ") is Not Valid, ";
+
+		if (validationMessage.equalsIgnoreCase("BuildingAddressCountryCode(")) {
+			if (!attributeValue.equalsIgnoreCase("USA") && attributeLen > maxAttributeLen) {
+				reportMsg += validationMessage.concat(attributeValue) + ") is longer than " + maxAttributeLen + ", ";
+				valid = false;
+			}
+		} 
+		else {
+			if (attributeLen > maxAttributeLen) {
+				reportMsg += validationMessage.concat(attributeValue) + ") is longer than " + maxAttributeLen + ", ";
+				valid = false;
+			}
 		}
+
 		if (!valid) {
 			writetoErrorReportOutputFile(outReportErrorWriter, reportMsg);
+			return valid;
 		}
 		return valid;
 	}
-	
+
 	private void updateBuilding(Building building, String routeCodeValue, String campusCode, String reportMsg, PrintWriter outReportWriter, ArchibusBuildings archBuilding, RouteCode routecode) {
 		BuildingExtension buildingExt;
 		buildingExt = (BuildingExtension) building.getExtension();
-		LOG.debug("Building exists so just saving building if there were changes");
```

Comment:
We should keep a debug statement present in this method in case there are failures.